test/WebJobs.Extensions.DurableTask.Analyzers.Test/Orchestrator/DependencyInjectionAnalyzerTests.cs

```diff
@@ -0,0 +1,161 @@
+            {
+                Id = DiagnosticId,
+                Message = string.Format(Resources.DeterministicAnalyzerMessageFormat, "injectedVariable"),
+                Severity = Severity,
+                Locations =
+                    new[] {
+                            new DiagnosticResultLocation("Test0.cs", 22, 34)
+                        }
+            };
+
+            expectedDiagnostics[1] = new DiagnosticResult
+            {
+                Id = DiagnosticId,
+                Message = string.Format(Resources.DeterministicAnalyzerMessageFormat, "injectedVariable"),
+                Severity = Severity,
+                Locations =
+                    new[] {
+                            new DiagnosticResultLocation("Test0.cs", 23, 64)
+                        }
+            };
+
+            VerifyCSharpDiagnostic(test, expectedDiagnostics);
+        }
+
+        [TestMethod]
+        public void DI_InjectedDifferentNameTest()
+        {
+            var test = @"
+    using System;
+    using System.Threading;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+
+    namespace VSSample
+    {
+        public class HelloSequence
+        {
+            private string diffVariableName;
+
+            public HelloSequence(string injectedVariable)
+            {
+                this.diffVariableName = injectedVariable;
+            }
+
+            [FunctionName(""DIAnalyzerTestCases"")]
+            public static async Task Run(
+            [OrchestrationTrigger] IDurableOrchestrationContext context)
+            {
+               var usingDIVar = diffVariableName;
+                var usingDISimpleMemberAccessExpression = this.diffVariableName;
+            }
+        }
+    }";
+            var expectedDiagnostics = new DiagnosticResult[2];
+            expectedDiagnostics[0] = new DiagnosticResult
+            {
+                Id = DiagnosticId,
+                Message = string.Format(Resources.DeterministicAnalyzerMessageFormat, "diffVariableName"),
+                Severity = Severity,
+                Locations =
+                    new[] {
+                            new DiagnosticResultLocation("Test0.cs", 22, 33)
+                        }
+            };
+
+            expectedDiagnostics[1] = new DiagnosticResult
+            {
+                Id = DiagnosticId,
+                Message = string.Format(Resources.DeterministicAnalyzerMessageFormat, "diffVariableName"),
+                Severity = Severity,
+                Locations =
+                    new[] {
+                            new DiagnosticResultLocation("Test0.cs", 23, 64)
+                        }
+            };
+
+            VerifyCSharpDiagnostic(test, expectedDiagnostics);
+        }
+
+        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+        {
+            return new DeterministicMethodAnalyzer();
+        }
+    }
```

Comment:
Can we add one more test with both an orchestrator and an activity function in the same class, but with the injected variable consumed only by the activity function (which should be legal)?  Something like this:

```csharp
public class HelloSequence
{
    private string injectedVariable;
    public HelloSequence(string injectedVariable)
    {
        this.injectedVariable = injectedVariable;
    }

    [FunctionName(""DIAnalyzerTestCases"")]
    public async Task Run([OrchestrationTrigger] IDurableOrchestrationContext context)
    {
        return context.CallActivityAsync<object>("DIAnalyzerTestActivity", null);
    }

    [FunctionName("DIAnalyzerTestActivity")]
    public async Task Run([ActivityTrigger] Object anything)
    {
        var usingDIVar = this.injectedVariable; // this should be fine
    }
}
```

Reply:
Great point, I'll add that test.